doc/user/kernels.rst

```diff
@@ -0,0 +1,100 @@
+Kernels
+=======
+Now we know how to allocate buffers, but we haven't actually run any code on
+the GPU. We'll start at a low level, and in later sections see how
+katsdpsigproc's higher-level abstractions can remove some of the drudgery of
+GPU programming.
+
+Like OpenCL, but unlike CUDA C/C++, GPU code is compiled on the fly as the
+program is running. A context has a :meth:`~.AbstractContext.compile` method to
+compile the code and return a :class:`.AbstractProgram`. At this level the
+code needs to be written in either CUDA or OpenCL, depending on which type of
+context you have. A program can consist of multiple kernels (functions written
+to be called from the host and run on the device); use
+:meth:`.AbstractProgram.get_kernel` to obtain the kernel with a specific
+name. Finally, use :meth:`.AbstractCommandQueue.enqueue_kernel` to run the
+code with arguments.
+
+Let's jump in and see our first full example of running code on the GPU. It
+will load random data into a buffer, triple every element, and transfer the
+results back to the host. It will only work on OpenCL.
+
+.. literalinclude:: examples/triple.py
+
+This is now a fair amount of code, although the first half of it uses
+functionality we've seen before. We're allocating an array of 50 floats, but
+padding it to 64 because we're using a work group size of 32 work items and 60
+is not a multiple of 32.
+
+When enqueueing the kernel, we pass a number of parameters:
+
+1. The kernel itself.
+2. A list of kernel arguments. Each argument corresponds to one of the
+   parameters of the :c:func:`!multiply` function in :const:`!SOURCE`. For
+   pointers we need to provide a buffer. We pass ``buf.buffer``, which is the
+   :class:`pyopencl.array.Array` that underpins the katsdpsigproc array. For
+   scalar arguments we need to use sized numpy scalars rather than Python
+   scalars, and they need to correspond to the type used in the kernel.
+3. The "global size", which is the total number (and shape) of work items to
+   run for this kernel. Note that this is similar to how OpenCL works, but a
```

Comment:
I'm not sure I understand the scope of your question, but this parameter must always be the number of work items rather than the number of blocks, even when the backend is CUDA.

Reply:
Yes, that was what I was trying to ask. Thanks.